yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -573,10 +573,19 @@ object ApplicationMaster extends Logging {
   }
 
   private[spark] def sparkContextInitialized(sc: SparkContext): Unit = {
+    if(master == null) {
```

Comment:
I'll just let you fix it on merge.

Also, thank you for attracting my attention to the launcher API.